src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java

```diff
@@ -135,8 +123,26 @@ public TopDocs topDocs(SearchContext context, FetchSubPhase.HitContext hitContex
             BitDocIdSetFilter parentFilter = context.bitsetFilterCache().getBitDocIdSetFilter(rawParentFilter);
             Filter childFilter = context.filterCache().cache(childObjectMapper.nestedTypeFilter(), null, context.queryParserService().autoFilterCachePolicy());
             Query q = new FilteredQuery(query, new NestedChildrenFilter(parentFilter, childFilter, hitContext));
-            context.searcher().search(q, topDocsCollector);
-            return topDocsCollector.topDocs(from(), size());
+
+            int topN = from() + size();
+            if (topN == 0) {
```

Comment:
I think you should just check size() == 0? If from is 3 and size 0, we still only need to count matches?